From 8f6deaf747131929841c138a559f5bc0120cc6f8 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Tue, 25 May 2021 04:41:48 -0700
Subject: [PATCH 17/23] octeontx2-af: Fix ending channel number calculation for
 SDP VF

commit 04a63d9ae98ff6a900cba94783e70d617722f0a7 from
git@git.assembla.com:cavium/WindRiver.linux.git

There is a logic to verify if the channel falls in the SDP channel
range. A minor mistake in calculating the ending channel was causing
failure for a case where we use all the SDP channels and last VF tries
to allocate the last channel. This patch fixes the mistake.

Signed-off-by: Veerasenareddy Burru <vburru@marvell.com>
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: I206a89cfa30a88d6d6993876651896504ec1d3f5
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52670
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 34e7496881ed..8b15e7119737 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -324,7 +324,7 @@ static int nix_interface_init(struct rvu *rvu, u16 pcifunc, int type, int nixlf,
 			for (vfid = 0; vfid < vf; vfid++)
 				req_chan_base += sdp_info->vf_rings[vfid];
 			req_chan_cnt = sdp_info->vf_rings[vf];
-			req_chan_end = req_chan_base + req_chan_cnt;
+			req_chan_end = req_chan_base + req_chan_cnt - 1;
 			if (req_chan_base < NIX_CHAN_SDP_CHX(0) ||
 			    req_chan_end > NIX_CHAN_SDP_CHX(255)) {
 				dev_err(rvu->dev,
-- 
2.31.1

